aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -204,6 +206,20 @@ SparseTensor new_with_dims_and_tensor_sparse_symint(
           /*version_counter=*/values.unsafeGetTensorImpl()->version_counter(),
           /*allow_tensor_metadata_change=*/true));
   alias_into_sparse(self, indices_shallow_copy, values_shallow_copy);
+  // alias_into_sparse overrides coalesced flag, so resetting the flag to
+  // the desired state here:
+  if (is_coalesced.has_value()) {
+    impl->set_coalesced(*is_coalesced);
+  }
+  // TODO: alias_into_sparse sets the coalesce flag to
+  // `self._values().shape[0] < 2`. There exist methods (e.g. permute
+  // on COO tensors when `dims[0] != 0` holds) that force coalesced
+  // flag to false even when nnz is less that 2. Here we cannot
+  // determine if this is the intention of such methods but it is
+  // likely that these methods are overly restrictive when estimating
+  // is_coalesced state. The condition `!is_coalesced && self._nnz() <
+  // 2` provides a way to detect and optimize such methods with
+  // respect to estimating the is_coalesced state.
```

Comment:
A thought: OOS for this change, but relevant to this TODO

Any method that currents flips coalesced to `false` could inherit the `nnz < 2` optimization if the logic were placed within `SparseTensorImpl::set_coalesced` instead of `SparseTensorImpl::set_nnz_and_narrow` where I originally put it. 

That could be confusing and/or dangerous though as `impl->set_coalesced(false)` would become silently equivalent to `impl->set_coalesed(nnz < 2)`

Reply:
In addition, I suggest that the following invariant always holds:
```
sparse_coo_tensor(..., is_coalesced=is_coalesced).is_coalesced() == is_coalesced
```
including tensors that have `nnz < 2`.

This property is useful when explicitly testing code pathways that operate on uncoalesced  tensors.